src/middlewared/middlewared/plugins/chart_releases_linux/host_path_attachments.py

```diff
@@ -1,6 +1,73 @@
+import os
+
 from middlewared.common.attachment import FSAttachmentDelegate
+from middlewared.service import private, Service
 from middlewared.utils.path import is_child
 
+from .utils import is_ix_volume_path, safe_to_ignore_path
+
+
+class ChartReleaseService(Service):
+
+    class Config:
+        namespace = 'chart.release'
+
+    @private
+    async def ix_mount_validation(self, path, path_type):
+        path_list = [p for p in path.split('/') if p.strip()]
+        if safe_to_ignore_path(path):
+            if path_list and len(path_list) < 3 and path_list == 'mnt':
+                return f'Invalid path {path}. Mounting root dataset or path outside a pool is not allowed'
+        elif path_list and path_list[0] == 'cluster':
+            if err := await self.middleware.call('chart.release.validate_cluster_path', path):
+                return err
+        else:
+            return f'{path!r} {path_type!r} not allowed to be mounted'
+
+    @private
+    async def attached_path_validation(self, path, path_type):
+        allowed_service_types = {
+            'Chart Releases',
+            'Rsync Task',
+            'Snapshot Task',
+            'Rsync Module',
+            'CloudSync Task',
+        }
+        in_use_attachments = []
+        for attachment_entry in filter(
+            lambda attachment: attachment['service'] not in allowed_service_types,
+            await self.middleware.call('pool.dataset.attachments_with_path', path)
+        ):
+            if attachment_entry['service'] == 'Kubernetes' and is_ix_volume_path(
+                path, (await self.middleware.call('kubernetes.config'))['dataset']
+            ):
+                continue
+
+            in_use_attachments.append(attachment_entry)
+
+        if in_use_attachments:
+            return f'Invalid mount {path!r} {path_type}. Following service(s) use this ' \
+                   f'path: {", ".join(attachment["type"] for attachment in in_use_attachments)}'
+
+    @private
+    async def validate_host_source_path(self, path):
+        paths = {
+            'path': path,
+        }
+        real_path = await self.middleware.run_in_thread(os.path.realpath, path)
+        if real_path != path:
+            paths[f'path (real path of {path})'] = real_path
+
+        for path_type, path_to_test in paths.items():
+            if err := await self.ix_mount_validation(path_to_test, path_type):
+                return err
+
+            if path_to_test.startswith('/mnt/'):
```

Comment:
Since we're apparently allowing cluster volumes, you need to validate whether SMB share is attached to it.